docs/Transport Manager Programming/index.md

```diff
@@ -1,567 +1,289 @@
-      * Working procedure:
-        * The worker waits for the connection of a mobile device
-        * When the connection request from any of the mobile devices arrives, TA establishes a connection (creates the data path) with this device and the connectoin listener sends a notification through the controller to the upper level with the device and application IDS
-    * Server Connection Factory
-      * Implements transport dependent connection that was originated by the user
-      * It is the developer's responbility to implement this worker
-      * If the transport does not support this ability, the worker may be absent
-      * Creates a connection with the device and the application specified by the user
-        * Both the device and the application must already be known to the TA at this moment
-        * TA may know about the device after the 'search' routine or after the 'restore previous state' routine
-        * If device is not known the Transport Adapter returns the error immediately
-      * When the connection is created the TA sends a notifications through the controller to the upper level
-  * Connection
-    * The main responsibility of the client and server connection works is to create a Connection
-    * Connection is the entity that is responsible for data transmitting between Core and the device
-    * Works and Connection are closely related
-    * Customing the connection
-      * Custom implementation of Connection must be used in custom worker(s) only
-      * It is not possible to use other types of data exchange in default workers
-      * It is possible to use the default Connection implementation in custom workers
-    * The default Connection implementation is based on sockets isolated in a separate thread
-    * Each transport specific worker shall use transport dependent initialization of the threaded socket connection
-    * If the default implementation is not convenient the developer can create his own Connection
-      * In this case custom workers shall also be created
-  * The Descriptor is used for manipulating devices and connections inside of the adapter
-
-#### Custom Transport Adapter creation
-
-Create a class that is derived from `TransportAdapterImpl` and add the implementation of necessary virtual methods.
+#### Device Scanner
 
-```
-#include "transport_adapter_impl.h"
-class MyTransportAdapter : public TransportAdapterImpl
-{
-protected:
-    virtual DeviceType getDeviceType() const;
-}
+The TCP Transport Adapter does not use a device scanner because it waits for incoming connections. We will use the Bluetooth Transport Adapter’s Device Scanner as an example here.
+
+The Init() function is called once and is responsible for preparing for the lifecycle of your device scanner. Here, the bluetooth device scanner will start the device scanner worker thread. This worker thread will either scan for devices repeatedly or only when requested via a conditional variable.
```

Comment:
Mention how the scan by request or auto scan is configured via the device scanner constructor

snippet from the bluetooth device scanner:

```
    , auto_repeat_search_(auto_repeat_search)
    , auto_repeat_pause_sec_(auto_repeat_pause_sec)
```